homeassistant/components/lcn/config_flow.py

```diff
@@ -106,4 +134,76 @@ async def async_step_import(self, data: ConfigType) -> ConfigFlowResult:
             self.hass.config_entries.async_update_entry(entry, data=data)
             return self.async_abort(reason="existing_configuration_updated")
 
-        return self.async_create_entry(title=f"{host_name}", data=data)
+        return self.async_create_entry(title=f"{data[CONF_HOST]}", data=data)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=USER_SCHEMA)
+
+        errors = None
+        if get_config_entry(self.hass, user_input):
+            errors = {CONF_BASE: "already_configured"}
+        elif (error := await validate_connection(user_input)) is not None:
+            errors = {CONF_BASE: error}
+
+        if errors is not None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.add_suggested_values_to_schema(
+                    USER_SCHEMA, user_input
+                ),
+                errors=errors,
+            )
+
+        data: dict = {
+            **user_input,
+            CONF_DEVICES: [],
+            CONF_ENTITIES: [],
+        }
+
+        return self.async_create_entry(title=data[CONF_HOST], data=data)
+
+
+class LcnOptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
Instead of an options flow, I think this should be a reconfigure flow (add a flow starting with `async_step_reconfigure` to the config flow)

Reply:
Thanks for the hint. I changed it to a reconfigure_step.